### PR TITLE
Fix versioning issue when `@added` is used implicitly and explicitly

### PR DESCRIPTION
**Pull request by tjprescott**
Fixes: https://github.com/microsoft/typespec/issues/3239

### CODE REVIEW

File: packages/versioning/src/versioning.ts [hunk 1]
```diff
@@ -782,6 +781,48 @@ export enum Availability {
...
+    parentMap = getAvailabilityMapInTimeline(program, type.model, timeline);
+  } else if (type.kind === "Operation" && type.interface !== undefined) {
+    parentMap = getAvailabilityMapInTimeline(program, type.interface, timeline);
+  }
+  if (parentMap === undefined) return undefined;
+  for (const [moment, availability] of parentMap.entries()) {
+    if (availability === Availability.Added) {
+      // FIXME: is this actually correct?
+      // We essentially want to return the first (earliest) time this is added.
+      return moment.versions().next().value;
```
Comment (tjprescott) [inline packages/versioning/src/versioning.ts:820]: @timotheeguerin is this logic actually correct? I'm not as familiar with how the timeline is supposed to work.

Comment (timotheeguerin) [reply packages/versioning/src/versioning.ts:820]: I don't think so all those version are the same moment, its just version for different libraries

File: packages/versioning/src/versioning.ts
Comment (timotheeguerin) [inline packages/versioning/src/versioning.ts:1]: Does looks like its fixed here 
https://cadlplayground.z22.web.core.windows.net/prs/3255/?c=aW1wb3J0ICJAdHlwZXNwZWMvaHR0cCI7CtIZcmVzdNUZdmVyc2lvbmluZyI7Cgp1c2luZyBUeXBlU3BlYy5WyR07CgpAxytlZCjHGHMpCkBzZXJ2aWNlKHsKICB0aXRsZTogIldpZGdldCBTxhsiLAp9KQpuYW1lc3BhY2UgRGVtb8cbOwoKZW51bSDIUCDERnYxLMQGMsUGMywKfQoKbW9kZWwgVGVzdMUixEw6IHN0cuYAmSAgLy8gVW5jb21tZW50IGFuZCBub3cgdjEgaXMgcmVuZHJlZCBjb3JyZWN0bHkuxjBAYWRk6wDKLnYxKQogIEByZW1vds0YMsUY0C4zxBbkAUXqAIh9Cg%3D%3D&e=%40typespec%2Fopenapi3&options=%7B%7D

Comment (timotheeguerin) [reply packages/versioning/src/versioning.ts:1]: sorry had a typo in my comment, I think its not fixed in the above playground `v1` doesn't have the type property

Comment (tjprescott) [reply packages/versioning/src/versioning.ts:1]: Got it. I'll add a test case. Thanks!

Comment (tjprescott) [reply packages/versioning/src/versioning.ts:1]: Hm... this was made trickier by the fact that whether we treat the parent containing no version information as "this thing was always available" depends on a lot of assumptions:

```
model Foo {
  @added(Versions.v2);
  a: string;
}

model Bar {
  @removed(Versions.v2);
  b: string;
}
```

For `Foo` we would _not_ want to interpret the fact that Foo has been available forever to override the fact that we explicitly say that `Foo.a` was added in v2. But with Bar, we need to do the exact opposite. Here it does need to inherit from the parent that b must have been available in v1 in order to be removed in v2.

I have a test case working, but it is an open question if we are creating a mess we one day won't be able to untangle.

**Review by timotheeguerin — APPROVED** [review]

**Pull request merged**